docs/LocalDevelopmentGuide.md

```diff
@@ -0,0 +1,223 @@
+# Local Development Guide
+
+So, you want to see some progress, preferably on Slack, and not just in the
+forms of unit testing? At this point, fear is actually a reasonable response.
+With this guide, you can be talking to your locally-hosted Slack bot in no time!
+
+> **Warning**: This only works smoothly with a Unix machine (macOS or Linux
+> variants). Windows users may be in for more pain than expected.
+
+## 1: Install ngrok
+
+Slack requires that all webhooks are passed through HTTPS. This is rather inconvenient
+if you just want to test while running on your local computer. Luckily, we have ngrok,
+a forwarding service that hosts a public HTTPS URL that passes to your local computer.
+Sign up for ngrok and download it [here][download-ngrok].
+
+After installing, run `ngrok http 5000` to create an ngrok URL that will be passed
+to your local port 5000. As long as you run Rocket on port 5000 (see below),
+you can then access it through the HTTPS URL that ngrok gives you. Note that it is
+very important to use the HTTPS URL, *not* the HTTP URL.
+
+## 2: Create a Slack Workspace
+
+For testing, it's useful to have your own Slack workspace set up. If you do not
+already have one, go [here][create-workspace] to create one, and follow the steps
+to set it up.
+
+## 3: Configure a Slack App
+
+Follow the link [here][make-slack-app] to create a new Slack app - you can name it
+whatever you like - and install it to the appropriate workspace.
+
+### Add a Bot User
+
+In "Add features and functionality", add a bot user. Since this is just for testing,
+you can name the bot user whatever you like.
+
+### Add Event Subscriptions
+
+In "Add features and functionality", add event subscriptions. In particular, under
```

Comment:
I suggest moving this after step 6: adding in credentials for `slack.toml`. Reason is because when you add a event subscription, it checks against the forwarded ngrok url at /slack/events but will get an error 500 since the app isn't running yet - thus blocking you from making the feature for the bot (as in you wont be able to save changes on the slack app).